src/main/java/org/jitsi/jigasi/xmpp/CallControlMucActivator.java

```diff
@@ -0,0 +1,422 @@
+            // sends initial stats, used some kind of advertising
+            // so jicofo can recognize us as real jigasi and load balance us
+            updatePresenceStatusForXmppProvider(pps);
+        }
+        catch (Exception e)
+        {
+            logger.error(e, e);
+        }
+
+        // getting direct access to the xmpp connection in order to add
+        // a listener for incoming iqs
+        if (pps instanceof ProtocolProviderServiceJabberImpl)
+        {
+            ((ProtocolProviderServiceJabberImpl) pps).getConnection()
+                .addPacketListener(this, this);
+        }
+    }
+
+    @Override
+    public void onJvbRoomJoined(GatewaySession source)
+    {
+        updatePresenceStatusForXmppProviders();
+    }
+
+    @Override
+    public void onSessionAdded(GatewaySession session)
+    {
+        session.setListener(this);
+    }
+
+    @Override
+    public void onSessionRemoved(GatewaySession session)
+    {
+        updatePresenceStatusForXmppProviders();
+    }
+
+    /**
+     * Gets the list of active sessions and update their presence in their
+     * control room.
+     * Counts number of active sessions as conference count, and number of
+     * all participants in all jvb rooms as global participant count.
+     */
+    private void updatePresenceStatusForXmppProviders()
+    {
+        SipGateway gateway = ServiceUtils.getService(
+            osgiContext, SipGateway.class);
+
+        int participants = 0;
+        for(GatewaySession ses : gateway.getActiveSessions())
+        {
+            participants += ses.getJvbChatRoom().getMembersCount();
+        }
+
+        Collection<ServiceReference<ProtocolProviderService>> refs
+            = ServiceUtils.getServiceReferences(
+            osgiContext,
+            ProtocolProviderService.class);
+
+        for (ServiceReference<ProtocolProviderService> ref : refs)
+        {
+            updatePresenceStatusForXmppProvider(
+                gateway, osgiContext.getService(ref), participants);
+        }
+    }
+
+    /**
+     * Sends <tt>ColibriStatsExtension</tt> as part of presence with
+     * load information for the <tt>ProtocolProviderService</tt> in its
+     * brewery room.
+     *
+     * @param gateway the <tt>SipGateway</tt> instance we serve.
+     * @param pps the protocol provider service
+     * @param participantsCount the participant count.
+     */
+    private void updatePresenceStatusForXmppProvider(
+        SipGateway gateway,
+        ProtocolProviderService pps,
+        int participantsCount)
+    {
+        if (ProtocolNames.JABBER.equals(pps.getProtocolName())
+            && pps.getAccountID() instanceof JabberAccountID
+            && !((JabberAccountID)pps.getAccountID()).isAnonymousAuthUsed())
+        {
+            try
+            {
+                String roomName = pps.getAccountID()
+                    .getAccountPropertyString(ROOM_NAME_ACCOUNT_PROP);
+                if (roomName == null)
+                {
```

Comment:
You are not alone. I've been moving { on new lines at some point in all of the code :)